packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -326,23 +333,26 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
         bytes32 depositHash = _getDepositHash(_depositData);
         RelayData storage relay = relays[depositHash];
 
-        // Note `hasPrice` will return false if liveness has not been passed in the optimistic oracle.
-        require(
-            relays[depositHash].relayState == RelayState.Pending &&
-                _getOptimisticOracle().hasPrice(
-                    address(this),
-                    bridgeAdmin.identifier(),
-                    relay.priceRequestTime,
-                    getRelayAncillaryData(_depositData, relay)
-                ),
-            "Relay not settleable"
+        // If relay was disputed, then parties in dispute need to go through OptimisticOracle to resolve dispute.
+        require(relays[depositHash].relayState == RelayState.Pending, "Relay not settleable");
+
+        // Attempt to settle OptimisticOracle price as a convenience for the slow relayer who will receive their
+        // dispute bond back. This doubles as a check that the optimistic price request has resolved because this will
+        // revert if the price has not resolved yet.
+        _getOptimisticOracle().settle(
```

Comment:
Sorry, this may be a dumb question, but where do we validate that the final price was 1e18 (i.e. the relay was valid)?

Reply:
despite this making _sence_ I think it might be worthwhile to double-check the value returned, even if it is just for making the code easier to reason about without adding any extra security/functionality.

Reply:
I removed `settle` and replaced it with a check to `settleAndGetPrice` which will settle the price if it is not resolved, revert if the price is not settleable, and return the resolved price.

Reply:
beautiful. looks good!